src/llvm/function.cr

```diff
@@ -83,4 +85,25 @@ struct LLVM::Function
   def delete
     LibLLVM.delete_function(self)
   end
+
+  def add_debug_param(debug_type)
+    debug_params << debug_type if debug_type
+  end
+
+  def naked?
+    (attributes & LLVM::Attribute::Naked) == LLVM::Attribute::Naked
+  end
+
+  def body_to_ll_string
+    String.build do |str|
+      str << "def " << self.name << "(#{self.params.join(", ")}):\n"
+      basic_blocks.each do |block|
+        str << "Block " << block.name << ":\n"
+        block.instructions.each do |inst|
+          str << "\t" << inst << " -> metadata: " << (inst.instruction_metadata || "N/A") << "\n"
+        end
+        str << "\n"
+      end
+    end
+  end
```

Comment:
Since this is only used for debugging, maybe we can leave it out?

Reply:
I can remove it. It is helpful for me for IR analysis but I always can get it back out of git history if needed or I can build a script that will parse IR file and will show that info in a more sane way with included source code lines for better understanding.

Reply:
Knowing the amount of ICE-related bugs we still have open, IMO it totally make sense to leave it for debugging purposes.

Reply:
It is not great tool as it seems. I have better idea to implement it as VSCode tool that will render LLVM IR code with all useful info under your mouse. It will give you much more power to analyze it than this output (Here you cannot see al hierarchy so it is way less useful)
.